apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx

```diff
@@ -204,28 +147,20 @@ export function StakeDialog({
                             selectedValidator={selectedValidator}
                             handleClose={handleClose}
                             onBack={handleBack}
-                            onStake={handleStake}
-                            gasBudget={newStakeData?.gasBudget}
-                            isTransactionLoading={isTransactionLoading}
+                            amountWithoutDecimals={amountWithoutDecimals}
+                            senderAddress={senderAddress}
+                            onSuccess={handleTransactionSuccess}
                         />
                     )}
                     {view === StakeDialogView.EnterTimelockedAmount && (
                         <EnterTimelockedAmountView
                             selectedValidator={selectedValidator}
                             maxStakableTimelockedAmount={maxStakableTimelockedAmount ?? BigInt(0)}
-                            hasGroupedTimelockObjects={groupedTimelockObjects.length > 0}
                             handleClose={handleClose}
                             onBack={handleBack}
-                            onStake={handleStake}
-                            gasBudget={newStakeData?.gasBudget}
-                            isTransactionLoading={isTransactionLoading}
-                        />
-                    )}
-                    {view === StakeDialogView.Unstake && stakedDetails && (
-                        <UnstakeView
-                            extendedStake={stakedDetails}
-                            handleClose={handleClose}
-                            showActiveStatus
+                            senderAddress={senderAddress}
+                            onSuccess={handleTransactionSuccess}
+                            amountWithoutDecimals={amountWithoutDecimals}
```

Comment:
something weird is happening here. I wanted to stake 12 iota, and only 1 is staked. At some point we broke the staking functionality

![image](https://github.com/user-attachments/assets/344be0e4-a8c4-4155-8c71-372b5d3e96e0)
![image](https://github.com/user-attachments/assets/95ac8392-7fdc-4a6d-accd-3645b909e8b5)